cirq/circuits/circuit.py

```diff
@@ -1570,22 +1574,26 @@ def to_text_diagram_drawer(
         diagram.write(0, 0, '')
         for q, i in qubit_map.items():
             diagram.write(0, i, qubit_namer(q))
-        if any(
-                isinstance(op, cirq.GlobalPhaseOperation)
-                for op in self.all_operations()):
-            diagram.write(0,
-                          max(qubit_map.values(), default=0) + 1,
-                          'global phase:')
+
+        # Print out global phase information
+        for op in self.all_operations():
+            if isinstance(op, cirq.GlobalPhaseOperation):
+                diagram.write(0,
```

Comment:
This appears to overwrite the entry of the diagram for each global phase in the circuit so if there are more than one, the diagram will only show the last.  How do you want to handle coalescing tags from multiple global phase ops?

I think what the original code did (but I haven't tested) is just add 'global phase:' here and then print the value of each global phase op under each moment.  You can just add tags there.

Reply:
Yeah, I didn't quite understand how this worked when I made the first commit.  Can you take another look?